ingestion/src/kldata.rs

```diff
@@ -0,0 +1,284 @@
+use crate::{
+    permissions::{timeseries_is_open, ParamPermitTable, StationPermitTable},
+    Datum, Error, PooledPgConn,
+};
+use chrono::{DateTime, Utc};
+use regex::Regex;
+use std::{
+    collections::HashMap,
+    fmt::Debug,
+    str::{FromStr, Lines},
+    sync::{Arc, RwLock},
+};
+
+pub struct ObsinnObs {
+    timestamp: DateTime<Utc>,
+    id: ObsinnId,
+    value: f32,
+}
+
+pub struct ObsinnChunk {
+    observations: Vec<ObsinnObs>,
+    station_id: i32, // TODO: change name here to nationalnummer?
+    type_id: i32,
+}
+
+#[derive(Debug, Clone)]
+struct ObsinnId {
+    param_code: String,
+    sensor_and_level: Option<(i32, i32)>,
+}
```

Comment:
Yes, individual observations from obsinn are only identified by paramcode and (optionally) sensor and level. station_id and type_id are only specified for the whole message, hence why they are in ObsinnChunk.

I can see that the choice of names here leaves something to be desired, I mostly cribbed the naming convention from kvkafka. If you have any suggestions for better names I would be very open to them

Reply:
Ok, well naming wise I think I like ObsinnObs - that makes sense since it contains the actual observation. 
ObsinnID is also ok. I think maybe its chunk I struggle with, but I get why you called it that. Its sort of a grouping of observations. Maybe a 'parcel' or 'group' more than a chunk? or something really specific about it being a stationid and time resolution sort of clustering...?

Reply:
The chunk thing I got directly from kvkafka, it's what Isak used to describe a bunch of related observations that come in together. I'm not sure "parcel" or "group" make it clearer, all three have pretty much the same meaning. Maybe a doc comment would do a better job of clarifying this?

Reply:
I added some doc comments for these types, as well as for the response format obsinn expects